src/libraries/System.Security.Cryptography/ref/System.Security.Cryptography.cs

```diff
@@ -1626,7 +1626,10 @@ public partial class PasswordDeriveBytes : System.Security.Cryptography.DeriveBy
         [System.Runtime.Versioning.SupportedOSPlatformAttribute("windows")]
         public byte[] CryptDeriveKey(string? algname, string? alghashname, int keySize, byte[] rgbIV) { throw null; }
         protected override void Dispose(bool disposing) { }
+        [System.Obsolete("Rfc2898DeriveBytes replaces PasswordDeriveBytes for deriving key material from a password and is preferred in new applications.")]
+#pragma warning disable 0809
```

Comment:
No it doesn't. @bartonjs applying the ObsoleteAttribute in the derived `PasswordDeriveBytes` but not in the base class `DeriveBytes` causes the compiler to emit a CS0809 warning. I don't think ObsoleteAttribute is designed to point a derived type's overridden API to another derived type's overridden API.

- Should we remove the ObsoleteAttribute from PasswordDeriveBytes in favor of another way to point users to the other type?
- Should the entire `PasswordDeriveBytes` type be marked as obsolete instead?

Reply:
Marking the override as obsolete was done in .NET Framework.  The type itself isn't entirely obsolete because it had two purposes... the obsoleted one and the Windows-only CryptDeriveKey method.

Personally, I'd just put CS0809 in the NoWarn list for all ref projects (or just manually on ones that encounter it).

Assuming that the ref.cs is generated from the/a product assembly, then the warning would be replicated there... where someone would either suppress it, or change their approach.

Reply:
Putting CS0809 onto that project works for me. Will do that, thanks.